src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexAssemblyCompiler.cs

```diff
@@ -0,0 +1,273 @@
+                        {
+                            Ldthisfld(capslistField);
+
+                            Ldc(i);
+                            Ldstr(tree.CapsList[i]);
+                            _ilg.Emit(OpCodes.Stelem_Ref);
+                        }
+                    }
+
+                    // set capsize
+                    Ldthis();
+                    Ldc(code.CapSize);
+                    Stfld(RegexField(nameof(Regex.capsize)));
+
+                    // set runnerref and replref by calling InitializeReferences()
+                    Ldthis();
+                    Call(typeof(Regex).GetMethod("InitializeReferences", BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic)!);
+
+                    Ret();
+                }
+
+                // Constructor with the timeout parameter:
+                // We only generate a constructor with a timeout parameter if the regex information supplied has a non-infinite timeout.
+                // If it has an infinite timeout, then the generated code is not going to respect the timeout.
+                if (matchTimeout != Regex.InfiniteMatchTimeout)
+                {
+                    _method = null;
+                    timeoutCtor = _type.DefineConstructor(MethodAttributes.Public, CallingConventions.Standard, new Type[] { typeof(TimeSpan) });
+                    _ilg = timeoutCtor.GetILGenerator();
+                    {
+                        // Call the default constructor:
+                        Ldthis();
+                        _ilg.Emit(OpCodes.Call, defaultCtor);
+
+                        // Validate timeout:
+                        _ilg.Emit(OpCodes.Ldarg_1);
+                        Call(typeof(Regex).GetMethod(nameof(Regex.ValidateMatchTimeout), BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic)!);
+
+                        // Set timeout:
+                        Ldthis();
+                        _ilg.Emit(OpCodes.Ldarg_1);
+                        Stfld(internalMatchTimeoutField);
+
+                        Ret();
+                    }
+                }
+            }
+
+            // bake the type
+            _type.CreateType();
+            _type = null;
+            _method = null;
+            _ilg = null;
+        }
+
+        private void GenerateInitTrackCount()
+        {
+            Ldthis();
+            Ldc(_trackcount);
+            Stfld(s_runtrackcountField);
+            Ret();
+        }
+
+        internal void GenerateCreateHashtable(FieldInfo field, Hashtable ht)
+        {
+            MethodInfo addMethod = typeof(Hashtable).GetMethod(nameof(Hashtable.Add), BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic)!;
+
+            Ldthis();
+            Newobj(typeof(Hashtable).GetConstructor(Type.EmptyTypes)!);
+
+            Stfld(field);
+
+            IDictionaryEnumerator en = ht.GetEnumerator();
```

Comment:
Foreach?

Reply:
This avoids boxing each entry; I assume that's why it was written like this.